src/main/java/io/kestra/plugin/nats/Request.java

```diff
@@ -0,0 +1,198 @@
+        title = "Timeout in milliseconds to wait for a response.",
+        description = "Defaults to 5000 ms."
+    )
+    @Builder.Default
+    private Duration requestTimeout = Duration.ofMillis(5000);
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        // 1) Connect to NATS
+        Connection connection = this.connect(runContext);
+
+        // 2) Interpolate the subject (if it has placeholders like {{ ... }})
+        String renderedSubject = runContext.render(this.subject);
+
+        // 3) Retrieve a single "message map" (headers + data)
+        Map<String, Object> messageMap = retrieveSingleMessage(runContext);
+
+        // 4) Build the NATS Message
+        Message natsMessage = buildRequestMessage(renderedSubject, messageMap);
+
+        // 5) Execute request-reply with the configured timeout
+        Duration timeoutDuration = this.requestTimeout;
+        Message reply = connection.request(natsMessage, timeoutDuration);
+
+        // 6) Convert the reply (if any) to a UTF-8 string
+        String response = (reply == null) ? null : new String(reply.getData(), StandardCharsets.UTF_8);
+
+        connection.close();
+
+        return Output.builder()
+            .response(response)
+            .build();
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> retrieveSingleMessage(RunContext runContext) throws Exception {
+        // CASE 1: "from" is a String
+        if (this.from instanceof String fromStr) {
+            // Render the string (in case it has {{ }} placeholders)
+            String renderedStr = runContext.render(fromStr);
+
+            // If it starts with kestra://, read entire file content into data
+            if (renderedStr.startsWith("kestra://")) {
+                URI fromUri = new URI(renderedStr);
+
+                if (!"kestra".equalsIgnoreCase(fromUri.getScheme())) {
+                    throw new IllegalArgumentException("Invalid 'from': must be a kestra:// URI or a plain string.");
+                }
+
+                try (InputStream is = runContext.storage().getFile(fromUri)) {
+                    String content = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+                    return Map.of("data", content);
+                }
+            } else {
+                // Otherwise, treat the string itself as data
+                return Map.of("data", renderedStr);
+            }
+        }
+
+        // CASE 2: "from" is a List
+        if (this.from instanceof List<?> fromList) {
+            if (fromList.size() != 1) {
+                throw new IllegalArgumentException("'from' list must contain exactly one item for request-reply.");
+            }
+            Object single = fromList.get(0);
+            if (!(single instanceof Map<?, ?>)) {
+                throw new IllegalArgumentException("'from' list's single item must be a map.");
+            }
+            return (Map<String, Object>) single;
+        }
```

Comment:
It seems useless to support a list if it's only a list with a single element.

Let's not support list for now.